File: isabelle.eclipse.launch/src/isabelle/eclipse/launch/IsabelleLaunchPlugin.scala
package isabelle.eclipse.launch

import org.eclipse.core.runtime.{IStatus, Status}
import org.eclipse.ui.plugin.AbstractUIPlugin


/** 
 * @author Andrius Velykis
 */
object IsabelleLaunchPlugin {

  // The shared instance
  private var instance: IsabelleLaunchPlugin = _
  def plugin = instance

  def log(status: IStatus) = plugin.getLog.log(status)

  /** Returns a new error `IStatus` for this plug-in.
    *
    * @param message    text to have as status message
    * @param exception  exception to wrap in the error `IStatus`
    * @return  the error `IStatus` wrapping the exception
    */
  def error(ex: Option[Throwable] = None, msg: Option[String] = None): IStatus = {
    
    // if message is not given, try to use exception's
    val message = msg orElse { ex.flatMap(ex => Option(ex.getMessage)) }
    
    new Status(IStatus.ERROR, plugin.pluginId, 0, message.orNull, ex.orNull);
  }
}

class IsabelleLaunchPlugin extends AbstractUIPlugin {

  IsabelleLaunchPlugin.instance = this
  
  // The plug-in ID
  def pluginId = "isabelle.eclipse.launch" //$NON-NLS-1$
  
}

File: isabelle.eclipse.launch/src/isabelle/eclipse/launch/IsabelleLaunchPlugin.java
package isabelle.eclipse.launch;

import org.eclipse.core.runtime.IStatus;
import org.eclipse.core.runtime.Status;
import org.eclipse.ui.plugin.AbstractUIPlugin;
import org.osgi.framework.BundleContext;

/**
 * The activator class controls the plug-in life cycle
 */
public class IsabelleLaunchPlugin extends AbstractUIPlugin {

	// The plug-in ID
	public static final String PLUGIN_ID = "isabelle.eclipse.launch"; //$NON-NLS-1$

	// The shared instance
	private static IsabelleLaunchPlugin plugin;
	
	/**
	 * The constructor
	 */
	public IsabelleLaunchPlugin() {
	}

	/*
	 * (non-Javadoc)
	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#start(org.osgi.framework.BundleContext)
	 */
	public void start(BundleContext context) throws Exception {
		super.start(context);
		plugin = this;
	}

	/*
	 * (non-Javadoc)
	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#stop(org.osgi.framework.BundleContext)
	 */
	public void stop(BundleContext context) throws Exception {
		plugin = null;
		super.stop(context);
	}

	/**
	 * Returns the shared instance
	 *
	 * @return the shared instance
	 */
	public static IsabelleLaunchPlugin getDefault() {
		return plugin;
	}
	
	/**
	 * Writes the message to the plug-in's log
	 * 
	 * @param message the text to write to the log
	 */
	public void log(String message, Throwable exception) {
		IStatus status = newErrorStatus(message, exception);
		getLog().log(status);
	}
	
	/**
	 * Returns a new <code>IStatus</code> for this plug-in
	 */
	public static IStatus newErrorStatus(String message, Throwable exception) {
		if (message == null) {
			message = ""; 
		}		
		return new Status(IStatus.ERROR, PLUGIN_ID, 0, message, exception);
	}

}
